### PR TITLE
[PM-8655] Update web app new item button

### PR DESCRIPTION
**Pull request by alec-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-8655?atlOrigin=eyJpIjoiYTljMTE3NWI0Njg3NDc4ZDk2MjBiOTVmNTFhNGU3Y2YiLCJwIjoiaiJ9

## 📔 Objective

Updates the web vault "new" button and related functionality:

- Updates the top "new" button style (rounded corners, thicker plus icon)
- Updates the dropdown menu for the top "new" button to show individual cipher types and adds divider between types and containers (folder, collection)
- Updates the onboarding "new item" link to "new login" (and directs user to dialog with "login" cipher selected)
- Updates the "no items new item" button styles (rounded corners, thicker plus icon)

## 📸 Screenshots

Screenshot 2024-07-31 at 2 26 55 PM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/vault/individual-vault/vault.component.html [hunk 1]
```diff
@@ -79,8 +83,9 @@
         bitButton
         (click)="addCipher()"
         *ngIf="filter.type !== 'trash'"
+        [ngClass]="{ 'tw-rounded-full': extensionRefreshEnabled$ | async }"
```
Comment (shane-melton) [inline apps/web/src/app/vault/individual-vault/vault.component.html:86]: ⛏️  We don't need to round the button ourselves, it'll be taken care of with the design teams style feature branch once merged.

File: apps/web/src/app/vault/individual-vault/vault-header/vault-header.component.html [hunk 2]
```diff
@@ -69,30 +69,85 @@
...
+      <ng-container [ngSwitch]="extensionRefreshEnabled">
+        <ng-container *ngSwitchCase="true">
+          <button
+            bitButton
+            buttonType="primary"
+            type="button"
+            [bitMenuTriggerFor]="addOptions"
+            id="newItemDropdown"
+            appA11yTitle="{{ 'new' | i18n }}"
+            class="tw-rounded-full"
```
Comment (shane-melton) [inline apps/web/src/app/vault/individual-vault/vault-header/vault-header.component.html:81]: ⛏️ We don't need to round the button ourselves, it'll be taken care of with the design teams style feature branch once merged and apply to all `bitButton`s automatically.

File: apps/web/src/app/vault/individual-vault/vault.component.html [hunk 3]
```diff
@@ -79,8 +83,9 @@
...
       >
-        <i class="bwi bwi-plus bwi-fw" aria-hidden="true"></i>
+        <i class="bwi bwi-plus-f bwi-fw" aria-hidden="true"></i>
```
Comment (shane-melton) [inline apps/web/src/app/vault/individual-vault/vault.component.html:88]: ❓ @danielleflinn It's pretty minor but do we want this change from `bwi-plus` to `bwi-plus-f` to be behind the feature flag too? (this is for the empty vault New Item button)

**Review by shane-melton — CHANGES_REQUESTED** [review]
Looks good, nice work! Just two small nits regarding button rounding that will eventually be taken care of by the design team's style feature branch once merged.

File: apps/web/src/app/vault/individual-vault/vault.component.html (hunk 3 above)
Comment (danielleflinn) [reply apps/web/src/app/vault/individual-vault/vault.component.html:88]: No preference from me; whatever is easier for you all

Comment (shane-melton) [reply apps/web/src/app/vault/individual-vault/vault.component.html:88]: Keeping it as is works for me, thanks!

File: apps/web/src/app/vault/individual-vault/vault-header/vault-header.component.html (hunk 2 above)
Comment (alec-livefront) [reply apps/web/src/app/vault/individual-vault/vault-header/vault-header.component.html]: 👍 [f9a2eef](https://github.com/bitwarden/clients/pull/10354/commits/f9a2eef924cc394eb12386536967a2ac5d1edb98)

File: apps/web/src/app/vault/individual-vault/vault.component.html (hunk 1 above)
Comment (alec-livefront) [reply apps/web/src/app/vault/individual-vault/vault.component.html]: 👍 [c816269](https://github.com/bitwarden/clients/pull/10354/commits/c816269f2eee7209b2a670462ec23a711bd7dd33)

**Review by shane-melton — APPROVED** [review]

**Pull request merged**